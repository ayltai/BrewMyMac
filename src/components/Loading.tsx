import { Backdrop, CircularProgress, } from '@mui/material';
import React from 'react';

export const Loading = () => (
    <Backdrop open>
        <CircularProgress color='inherit' />
    </Backdrop>
);
