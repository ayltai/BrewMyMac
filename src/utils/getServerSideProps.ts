import { serverSideTranslations, } from 'next-i18next/serverSideTranslations';
import nextI18NextConfig from '../../next-i18next.config';

export default async function getServerSideProps({
    locale,
} : {
    locale : string,
}) {
    return {
        props : {
            ...(await serverSideTranslations(locale, [
                'common',
            ], nextI18NextConfig)),
        },
    };
}
