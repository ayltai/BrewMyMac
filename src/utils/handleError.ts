import * as Sentry from '@sentry/nextjs';

export const handleError = (error : any) => {
    if (process.env.NODE_ENV === 'production') {
        Sentry.captureException(error);
    } else {
        console.error(error);
    }
};
