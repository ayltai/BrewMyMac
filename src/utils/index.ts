export { createEmotionCache, } from './createEmotionCache';
export { handleError, } from './handleError';
