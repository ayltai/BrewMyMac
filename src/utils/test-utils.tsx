import { createTheme, ThemeProvider, } from '@mui/material/styles';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { act, render, } from '@testing-library/react';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { homebrewApi, } from '../apis';
import { preferencesReducer, shoppingCartReducer, store as reduxStore, } from '../states';
import { palette, } from '../styles';

import type { HomebrewFormula, } from '../types';

const INITIAL_FILTER = 'firefox';

export const mockApis = async () => {
    const HOMEBREW_API_RESPONSE : HomebrewFormula[] = [
        {
            token     : `${INITIAL_FILTER} homebrew token`,
            name      : [
                `${INITIAL_FILTER} homebrew name`,
            ],
            full_name : `${INITIAL_FILTER} homebrew full name`,
            bottle    : {},
        },
    ];

    const HOMEBREW_CAST_API_RESPONSE : HomebrewFormula[] = [
        {
            token     : `${INITIAL_FILTER} homebrew cask token`,
            name      : [
                `${INITIAL_FILTER} homebrew cask name`,
            ],
            full_name : `${INITIAL_FILTER} homebrew cask full name`,
            bottle    : {},
        },
    ];

    fetchMock.doMock(request => {
        if (request.url.startsWith('https://formulae.brew.sh/api/formula.json')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(HOMEBREW_API_RESPONSE),
            });
        }

        if (request.url.startsWith('https://formulae.brew.sh/api/cask.json')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(HOMEBREW_CAST_API_RESPONSE),
            });
        }

        return Promise.resolve({
            status : 404,
        });
    });

    await act(async () => {
        await reduxStore.dispatch(homebrewApi.endpoints.formula.initiate());
        await reduxStore.dispatch(homebrewApi.endpoints.cask.initiate());
    });
};

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences  : preferencesReducer,
        shoppingCart : shoppingCartReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : PreloadedState<RootState>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => {
        const theme = createTheme({
            palette : {
                ...palette,
                mode : 'light',
            },
        });

        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    },
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
