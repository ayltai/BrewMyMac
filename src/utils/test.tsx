import { createTheme, ThemeProvider, } from '@mui/material';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { act, render, } from '@testing-library/react';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { appStoreApi, homebrewApi, tweakApi, } from '../apis';
import type { AppStoreSearchResults, HomebrewFormula, Tweak, } from '../models';
import { sessionReducer, store as reduxStore, } from '../redux';
import { palette, } from '../styles';

const INITIAL_FILTER = 'firefox';

export const mockApis = async () => {
    const HOMEBREW_API_RESPONSE : HomebrewFormula[] = [
        {
            token     : `${INITIAL_FILTER} homebrew token`,
            name      : [
                `${INITIAL_FILTER} homebrew name`,
            ],
            full_name : `${INITIAL_FILTER} homebrew full name`,
        },
    ];

    const HOMEBREW_CAST_API_RESPONSE : HomebrewFormula[] = [
        {
            token     : `${INITIAL_FILTER} homebrew cask token`,
            name      : [
                `${INITIAL_FILTER} homebrew cask name`,
            ],
            full_name : `${INITIAL_FILTER} homebrew cask full name`,
        },
    ];

    const TWEAKS_API_RESPONSE : Tweak[] = [
        {
            id   : '2',
            name : `${INITIAL_FILTER} tweak name`,
        },
    ];

    const APP_STORE_API_RESPONSE : AppStoreSearchResults = {
        resultCount : 1,
        results     : [
            {
                trackId   : '1',
                trackName : `${INITIAL_FILTER} app store name`,
            },
        ],
    };

    fetchMock.doMock(request => {
        if (request.url.startsWith('https://formulae.brew.sh/api/formula.json')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(HOMEBREW_API_RESPONSE),
            });
        }

        if (request.url.startsWith('https://formulae.brew.sh/api/cask.json')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(HOMEBREW_CAST_API_RESPONSE),
            });
        }

        if (request.url.startsWith('https://raw.githubusercontent.com')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(TWEAKS_API_RESPONSE),
            });
        }

        if (request.url.startsWith('https://search-itunes.vercel.app')) {
            return Promise.resolve({
                status : 200,
                body   : JSON.stringify(APP_STORE_API_RESPONSE),
            });
        }

        return Promise.resolve({
            status : 404,
        });
    });

    await act(async () => {
        await reduxStore.dispatch(homebrewApi.endpoints.formula.initiate());
        await reduxStore.dispatch(homebrewApi.endpoints.cask.initiate());
        await reduxStore.dispatch(tweakApi.endpoints.getTweaks.initiate());
        await reduxStore.dispatch(appStoreApi.endpoints.search.initiate(`${INITIAL_FILTER} app store name`));
    });
};

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        session : sessionReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : PreloadedState<RootState>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => {
        const theme = createTheme({
            palette : {
                ...palette,
                mode : 'light',
            },
        });

        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    },
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
