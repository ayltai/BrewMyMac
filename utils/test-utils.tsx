import { createTheme, ThemeProvider, } from '@mui/material/styles';
import { configureStore, PreloadedState, Store, } from '@reduxjs/toolkit';
import '@testing-library/jest-dom/extend-expect';
import { render, } from '@testing-library/react';
import React, { ReactNode, } from 'react';
import { Provider, } from 'react-redux';

import { preferencesReducer, } from '../states';
import { palette, } from '../styles';

const createStore = (preloadedState? : Record<string, any>) => configureStore({
    reducer : {
        preferences : preferencesReducer,
    },
    preloadedState,
});

const defaultStore = createStore();

type RootState = ReturnType<typeof defaultStore.getState>;

const customRender = (ui : any, {
    preloadedState,
    store = createStore(preloadedState),
    ...rest
} : {
    preloadedState?   : PreloadedState<RootState>,
    store?            : Store<RootState>,
    [ rest : string ] : any,
} = {}) => render(ui, {
    wrapper : ({
        children,
    } : {
        children : ReactNode,
    }) => {
        const theme = createTheme({
            palette : {
                ...palette,
                mode : 'light',
            },
        });

        return (
            <Provider store={store}>
                <ThemeProvider theme={theme}>
                    {children}
                </ThemeProvider>
            </Provider>
        );
    },
    ...rest,
});

export * from '@testing-library/react';

export { customRender as render, };
