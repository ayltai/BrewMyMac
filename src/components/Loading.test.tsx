import React from 'react';

import { render, } from '../utils/test';

import { Loading, } from './Loading';

describe('<ExemptionRecord />', () => {
    it('renders correctly', () => expect(render(<Loading />).asFragment()).toMatchSnapshot());
});
