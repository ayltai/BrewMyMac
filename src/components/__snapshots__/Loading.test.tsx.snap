// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExemptionRecord /> renders correctly 1`] = `
<DocumentFragment>
  <div
    aria-hidden="true"
    class="MuiBackdrop-root css-sox5kk-MuiBackdrop-root"
    style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
  >
    <span
      class="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorInherit css-62e83j-MuiCircularProgress-root"
      role="progressbar"
      style="width: 40px; height: 40px;"
    >
      <svg
        class="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          class="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx="44"
          cy="44"
          fill="none"
          r="20.2"
          stroke-width="3.6"
        />
      </svg>
    </span>
  </div>
</DocumentFragment>
`;
